availability/watcher.py

```diff
@@ -80,21 +73,27 @@ def save_availability(results_queue):
         results.append(data)
 
     body = []
+    indices = set()
     for data in results:
-        metadata = {"index": {"_index": "ms_availability_%(region)s" % data,
+        index = "ms_availability_%(region)s" % data
+        metadata = {"index": {"_index": index,
                               "_type": "service_availability",
                               "_id": str(uuid.uuid1())}}
         body.append(json.dumps(metadata, indent=0).replace("\n", ""))
         body.append("\n")
         body.append(json.dumps(data, indent=0).replace("\n", ""))
         body.append("\n")
+        if index not in indices:
+            storage.check_create_elasticsearch_index(index)
+            indices.add(index)
     body = "".join(body)
 
-    es = elasticsearch.Elasticsearch([backend["connection"]])
+    es = storage.get_elasticsearch()
+
     LOG.debug("Saving availability:\n%s" % body)
     try:
         es.bulk(body=body)
-    except Exception as e:
+    except storage.elasticsearch.exceptions.ElasticsearchException as e:
```

Comment:
please import here elasticsearch excpetions module and use it instead of using it from storage this is really bad pattern.